backend/app/services/price_history_manager.py

```
@@ -0,0 +1,380 @@
+
+                # Skip if we already have this data and not forcing update
+                if not force_update and price_date in existing_dates:
+                    skipped += 1
+                    continue
+
+                # Convert to EUR if needed
+                price_usd = price_data['close']
+                try:
+                    usd_to_eur_rate = get_exchange_rate("USD", "EUR")
+                    price_eur = price_usd * usd_to_eur_rate
+                except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)